runtime/vm/description.c

```diff
@@ -164,14 +164,19 @@ calculateInstanceDescription( J9VMThread *vmThread, J9Class *ramClass, J9Class *
 	 * set the corresponding description bit for each 
 	 */
 	{
+		/* Check if there are self referencing field offsets being inherited.
+		 * If so, self referencing fields of the derived class should be ignored.
+		 */
+		BOOLEAN inheritedSelfReferencingFields = ramClass->selfReferencingField1 != 0 ? TRUE : FALSE;
```

Comment:
I'd add.... or in other words both self referencing fields must be from the same class.